File: solutions/c/rotational-cipher/1/rotational_cipher.c
#include "rotational_cipher.h"
#include <string.h>
#include <stdlib.h>

char *rotate(const char *text, int shift_key) {
    size_t n = strlen(text);
    char *result = (char*)malloc(sizeof(char) * n);
    for (size_t i = 0; i < n; ++i) {
        if (text[i] >= 'A' && text[i] <= 'Z') {
        	result[i] = (text[i] - 'A' + shift_key) % 26 + 'A';
        } else if (text[i] >= 'a' && text[i] <= 'z') {
        	result[i] = (text[i] - 'a' + shift_key) % 26 + 'a';
        } else {
            result[i] = text[i];
        }
    }

    return result;
}

File: solutions/go/rotational-cipher/1/rotational_cipher.go
package rotationalcipher

func RotationalCipher(plain string, shiftKey int) string {
	data := []rune(plain)
    for i := 0; i < len(data); i++ {
        if data[i] >= 'A' && data[i] <= 'Z' {
        	data[i] = (data[i] - 'A' + rune(shiftKey)) % 26 + 'A'
        }
        if data[i] >= 'a' && data[i] <= 'z' {
        	data[i] = (data[i] - 'a' + rune(shiftKey)) % 26 + 'a'
        }
    }

    return string(data)
}
